### DISCUSSION TITLE
'failed to create process error' after installation

### DISCUSSION THREAD

**Discussion by kechegoyen** [talmolab/sleap#791] [Help!] [answered]
# Bug description

I was able to download and install all of the packages for SLEAP, but my GUI will not pop-up when I type the command "sleap-label".

## Expected behaviour
GUI opens. 

## Actual behaviour
Failed to create process. 

## Your personal set up

SLEAP: 1.2.3
TensorFlow: 2.6.3
Numpy: 1.19.5
Python: 3.7.12
OS: Windows-10-10.0.22000-SP0

I actually have Windows 11. I am not sure why it says Windows 10 above. 

GPUs: 1/1 available
  Device: /physical_device:GPU:0
         Available: True
        Initalized: False
     Memory growth: None
[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]

 - SLEAP installation method (listed [here](https://sleap.ai/installation.html#)):
    * [ ] [Conda from package](https://sleap.ai/installation.html#conda-package)

<details><summary>Environment packages</summary>

```
# paste output of `pip freeze` or `conda list` here
```
# packages in environment at C:\Users\Kimberly Echegoyen\.conda\envs\sleap:
#
# Name                    Version                   Build  Channel
absl-py                   0.15.0                   pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
attrs                     21.2.0                   pypi_0    pypi
backports-zoneinfo        0.2.1                    pypi_0    pypi
ca-certificates           2022.5.18.1          h5b45459_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cachetools                4.2.4                    pypi_0    pypi
cattrs                    1.1.1                    pypi_0    pypi
certifi                   2021.10.8                pypi_0    pypi
charset-normalizer        2.0.12                   pypi_0    pypi
clang                     5.0                      pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
commonmark                0.9.1                    pypi_0    pypi
cuda-nvcc                 11.3.58              hb8d16a4_0    nvidia
cudatoolkit               11.3.1              h280eb24_10    conda-forge
cudnn                     8.2.1.32             h754d62a_0    conda-forge
cycler                    0.11.0                   pypi_0    pypi
efficientnet              1.0.0                    pypi_0    pypi
ffmpeg                    4.3.1                ha925a31_0    conda-forge
flatbuffers               1.12                     pypi_0    pypi
fonttools                 4.33.3                   pypi_0    pypi
freetype                  2.10.4               h546665d_1    conda-forge
gast                      0.4.0                    pypi_0    pypi
geos                      3.9.1                h39d44d4_2    conda-forge
google-auth               1.35.0                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.44.0                   pypi_0    pypi
h5py                      3.1.0           nompi_py37h19fda09_100    conda-forge
hdf5                      1.10.6          nompi_he0bbb20_101    conda-forge
idna                      3.3                      pypi_0    pypi
image-classifiers         1.0.0                    pypi_0    pypi
imageio                   2.15.0                   pypi_0    pypi
imgaug                    0.4.0                    pypi_0    pypi
imgstore                  0.2.9                    pypi_0    pypi
importlib-metadata        4.11.1                   pypi_0    pypi
intel-openmp              2022.1.0          h57928b3_3787    conda-forge
joblib                    1.1.0                    pypi_0    pypi
jpeg                      9e                   h8ffe710_1    conda-forge
jsmin                     3.0.1                    pypi_0    pypi
jsonpickle                1.2                      pypi_0    pypi
keras                     2.6.0                    pypi_0    pypi
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.2                    pypi_0    pypi
lcms2                     2.12                 h2a16943_0    conda-forge
lerc                      3.0                  h0e60522_0    conda-forge
libblas                   3.9.0              15_win64_mkl    conda-forge
libcblas                  3.9.0              15_win64_mkl    conda-forge
libdeflate                1.10                 h8ffe710_0    conda-forge
liblapack                 3.9.0              15_win64_mkl    conda-forge
libpng                    1.6.37               h1d00b33_2    conda-forge
libtiff                   4.3.0                hc4061b1_4    conda-forge
libzlib                   1.2.12               h8ffe710_0    conda-forge
lz4-c                     1.9.3                h8ffe710_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markdown                  3.3.6                    pypi_0    pypi
matplotlib                3.5.2                    pypi_0    pypi
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
networkx                  2.6.3                    pypi_0    pypi
numpy                     1.19.5           py37h4c2b6ed_3    conda-forge
oauthlib                  3.2.0                    pypi_0    pypi
olefile                   0.46               pyh9f0ad1d_1    conda-forge
opencv-python             4.5.5.62                 pypi_0    pypi
opencv-python-headless    4.5.5.62                 pypi_0    pypi
openjpeg                  2.4.0                hb211442_1    conda-forge
openssl                   3.0.3                h8ffe710_0    conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pandas                    1.3.5            py37h9386db6_0    conda-forge
pillow                    8.4.0            py37hd7d9ad0_0    conda-forge
pip                       22.0.3             pyhd8ed1ab_0    conda-forge
protobuf                  3.19.4                   pypi_0    pypi
psutil                    5.9.0                    pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pygments                  2.12.0                   pypi_0    pypi
pykalman                  0.9.5                    pypi_0    pypi
pyparsing                 3.0.6                    pypi_0    pypi
pyreadline                2.1             py37h03978a9_1005    conda-forge
pyside2                   5.14.1                   pypi_0    pypi
python                    3.7.12          h900ac77_100_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-rapidjson          1.6                      pypi_0    pypi
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
pytz-deprecation-shim     0.1.0.post0              pypi_0    pypi
pywavelets                1.3.0                    pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qimage2ndarray            1.8.3                    pypi_0    pypi
requests                  2.27.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rich                      10.16.1                  pypi_0    pypi
scikit-image              0.19.2                   pypi_0    pypi
scikit-learn              1.0.2                    pypi_0    pypi
scikit-video              1.1.11                   pypi_0    pypi
scipy                     1.7.3            py37hb6553fb_0    conda-forge
seaborn                   0.11.2                   pypi_0    pypi
segmentation-models       1.0.1                    pypi_0    pypi
setuptools                59.8.0           py37h03978a9_1    conda-forge
setuptools-scm            6.4.2                    pypi_0    pypi
shapely                   1.7.1            py37hc520ffa_5    conda-forge
shiboken2                 5.14.1                   pypi_0    pypi
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sleap                     1.2.3                    pypi_0    pypi
sqlite                    3.38.5               h8ffe710_0    conda-forge
tbb                       2021.5.0             h2d74725_1    conda-forge
tensorboard               2.6.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.6.3                    pypi_0    pypi
tensorflow-estimator      2.6.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2021.11.2                pypi_0    pypi
tk                        8.6.12               h8ffe710_0    conda-forge
tomli                     2.0.1                    pypi_0    pypi
typing-extensions         3.10.0.2                 pypi_0    pypi
tzdata                    2022.1                   pypi_0    pypi
tzlocal                   4.2                      pypi_0    pypi
ucrt                      10.0.20348.0         h57928b3_0    conda-forge
urllib3                   1.26.8                   pypi_0    pypi
vc                        14.2                 hb210afc_6    conda-forge
vs2015_runtime            14.29.30037          h902a5da_6    conda-forge
werkzeug                  2.0.3                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
wrapt                     1.12.1                   pypi_0    pypi
xz                        5.2.5                h62dcd97_1    conda-forge
zipp                      3.7.0                    pypi_0    pypi
zlib                      1.2.12               h8ffe710_0    conda-forge
zstd                      1.5.2                h6255e5f_1    conda-forge

<details><summary>Logs</summary>

```
(sleap) PS C:\Users\Kimberly Echegoyen> sleap-label
failed to create process.
(sleap) PS C:\Users\Kimberly Echegoyen> sleap-diagnostic
failed to create process.
```
</details>

## How to reproduce

(base) PS C:\Users\Kimberly Echegoyen> conda activate sleap
(sleap) PS C:\Users\Kimberly Echegoyen> sleap-label
failed to create process.

**Comment by talmo** [comment]
Hi @kechegoyen,

Converting this to a **Discussion** since it comes up fairly regularly. Paraphrasing from a [previous issue](https://github.com/talmolab/sleap/issues/594#issuecomment-946063628):

> This is an issue with the Python installation path. This is a common issue on Windows where you'll have problems if there are any spaces in the path to `python.exe`.
> 
> You can check by just trying to run python (should give you the same error) or by typing `where python` in the commandline and seeing the filename that gets printed.
> 
> The fix is to just reinstall Miniconda in `C:\Miniconda3` or `C:\ProgramFiles\Miniconda3`.


Let us know if that still doesn't work for you after reinstalling.

Cheers,

Talmo

**Comment by roomrys** [comment, marked as answer]
I recently helped a user correct this on their system, and while it is likely not the best solution, we got it working by doing the following:

0. [Uninstall miniconda](https://docs.anaconda.com/anaconda/install/uninstall/?highlight=uninstall)
1. Install miniconda to C:/miniconda3 with the following options selected:
- install for all users
- "Register Anaconda as my default Python"
2. [Install sleap](https://sleap.ai/installation.html#conda-package) in the anaconda terminal using a modified version of the usual command:
- remove the "-n sleap" 
- add a new [path to store the environment](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#specifying-a-location-for-an-environment) (--prefix <path_without_spaces>, i.e. `--prefix C:/Users/Public/sleap`)

```
conda create -y -c sleap -c nvidia -c conda-forge sleap --prefix C:/Users/Public/sleap
```
By this point things were working. For ease of use, we set the path to the sleap environment as an environment variable (accessible through `%<name_of_env_variable>%`):

3. Set the path to the conda environment as an environment variable `setx sleap C:/Users/Public/sleap`
4. Open a new Anaconda terminal! and activate the environment with `conda activate %sleap%`

---

There has to be a way to change where conda is storing environments by default. This post does not include that solution (which is probably the best solution), but proves that where the conda environment is being stored also matters (no spaces)!
